### ISSUE TITLE
Export data for underline 

### ISSUE THREAD

**Issue by kisvegabor** [lvgl/lv_font_conv#40]
Based on https://github.com/lvgl/lvgl/issues/1618#issuecomment-651403534

FreeType can tell where to place the underline of a text and how thick it should be. See [here](
https://www.freetype.org/freetype2/docs/tutorial/step2.html).

It'd be great to add this info to the font descriptors like this:
```c

/*Initialize a public general font descriptor*/
lv_font_t lv_font_montserrat_12 = {
    .get_glyph_dsc = lv_font_get_glyph_dsc_fmt_txt,    /*Function pointer to get glyph's data*/
    .get_glyph_bitmap = lv_font_get_bitmap_fmt_txt,    /*Function pointer to get glyph's bitmap*/
    .line_height = 15,          /*The maximum line height required by the font*/
    .base_line = 3,             /*Baseline measured from the bottom of the line*/
#if !(LVGL_VERSION_MAJOR == 6 && LVGL_VERSION_MINOR == 0)
    .subpx = LV_FONT_SUBPX_NONE,
#endif
#if LV_VERSION_CHECK(7, 2, 0)
    .underline_ofs = 3,
    .underline_tickness = 2,
#endif
    .dsc = &font_dsc           /*The custom font data. Will be accessed by `get_glyph_bitmap/dsc` */
};

```

**Comment by embeddedt** [comment]
How does this work when fonts are mixed? Do we just use the values from the first font (like the other metrics)?

**Comment by kisvegabor** [comment]
Good question. 
Using the first font (which contains this information) is a possible way.

And what to do if none of the fonts contain this info. Initialize  `.underline_ofs = 0` and `.underline_tickness = line_height / 10`?
(the 1/10 width is the current value and it produces good looking width in general)

**Comment by embeddedt** [comment]
If there's multiple fonts, I'm not sure if taking the average of the values might give better visual results. What do you think?

If none of the fonts contain the info, I think using defaults makes sense.

**Comment by puzrin** [comment]
> How does this work when fonts are mixed? Do we just use the values from the first font (like the other metrics)?

Yes, metrics of first font will be prefered. The same for ascent/descent and so on.

Font mixing is to create new fonts from icons, or to add some icons into font. If anyone wish to merge different text fonts - good luck :)

**Comment by kisvegabor** [comment]
>  The same for ascent/descent and so on.

Only `line_height` and `base_line` should be calculated from all the fonts to be sure every font to be sure all the glyphs fit into that line. But I suppose it's the case now as well.

**Comment by incity** [comment]
please also think about the font flip, italic, bold..

**Comment by puzrin** [comment]
@incity request to "think" is no concrete. Please specify exact properties of TTF font, according to OTF spec, and reasons.

**Comment by incity** [comment]
@puzrin like the CSS, there is a font-weight, font-size, font-style.

**Comment by puzrin** [comment]
@incity reasons (use case samples)? Size already exists. Properties can not be added "just for fun".

**Comment by incity** [comment]
![image](https://user-images.githubusercontent.com/7846605/86226727-a682ed00-bbbe-11ea-96ce-50fcb3903b77.png)
like this, you may want to make your title more prominent, so you need to set the font-weight to make your title font more bold.

**Comment by puzrin** [comment]
You'd worth to read font spec first. This is simplified bitmap font format, it does NOT support variations. If you need bold - you have to create separate bold font and assign it where needed.

**Comment by incity** [comment]
the freetype has the interface: FT_Bitmap_Embolden and FT_Set_Transform.

**Comment by puzrin** [comment]
This is not freetype. This is simplified format for easy render on small MCU-s.

**Comment by kisvegabor** [comment]
@puzrin Is there anything to discuss about underline?

**Comment by puzrin** [comment]
No, doing another things. Sorry for delay.

**Comment by kisvegabor** [comment]
No problem. Not pushing just asking :slightly_smiling_face:

**Comment by juice-man** [comment]
@kisvegabor also the pitch

**Comment by puzrin** [comment]
@juice-man default policy is "reject everything". If you think something important missed - please create more descriptive request, with link to ttf origin and explanation why demanded property is important. Also, use separate issue for each feature.

**Comment by puzrin** [comment]
Ref: https://docs.microsoft.com/en-us/typography/opentype/spec/post

Seems `post.underlinePosition` & `post.underlineThickness` needed.

**Comment by puzrin** [comment]
@kisvegabor could you try dev branch?

https://github.com/lvgl/lv_font_conv/commit/1d99544b6ab145311117550566f7fed10ddd21fb

**Comment by kisvegabor** [comment]
I've tested and works well. Only two things:
- `.underline_tickness`: a 'h' is missing. Should be`underline_thickness`
- `#if LV_VERSION_CHECK(7, 2, 0)` should be `#if LV_VERSION_CHECK(7, 4, 0)`

**Comment by puzrin** [comment]
https://github.com/lvgl/lv_font_conv/commit/f5cab4b90e8b335fe254c6dcf23b116af247009a

Force-pushed.

When you confirm everything is ok, i will merge & release.

**Comment by kisvegabor** [comment]
Looks good!

**Comment by puzrin** [comment]
released 1.1.0
